### DISCUSSION TITLE
Support for H3HexagonLayer

### DISCUSSION THREAD

**Discussion by assaflavi** [apache/superset#18643] [Ideas]
Migrated from https://github.com/apache-superset/superset-ui-plugins-deckgl/issues/29

> **Is your feature request related to a problem? Please describe.**
>
> I have a bunch of geospatial data indexed by H3 that I want to visualize. Currently I'm forced to export it into a CSV, and load it into an ephemeral session on https://kepler.gl/demo
>
> **Describe the solution you'd like**
> 
> I'd like this plugin to enable SuperSet visualizations using that type of layer, as it is already supported in deck.gl.
>
> **Describe alternatives you've considered**
>
> I'm not aware of any existing application that can do this.

**Comment by srinify** [comment]
Thanks! Moving to discussions (we're hoping to keep Github issues focused on bugs only)

**Comment by riazarbi** [comment]
I'd like to echo this need. I did so in the original issue but the migration from the old repo to discussions dropped my comment. 

We use h3 indexes extensively for warehousing of spatial data. Our existing workaround is to join in the geoJSON polygons and centroids to our tables before loading into Superset, but this really slows down our rendering. Including native support for H3HexagonLayer would be a major asset for us.

**Comment by JohnDietrich-Pepper** [comment]
Would like to see this as well

**Comment by riazarbi** [comment]
That is correct. What is needed is someone with the requisite skill to put in the time to integrate it.On 01 Feb 2023, at 02:59, JohnDietrich-Pepper ***@***.***> wrote:﻿
It looks like Deck.gl supports this but it isn’t exposed in Superset?  https://deck.gl/docs/api-reference/geo-layers/h3-hexagon-layer

**Reply by JohnDietrich-Pepper** [reply]
It seems like it wouldn’t take much to add a layer here. 

https://github.com/apache/superset/tree/master/superset-frontend/plugins/legacy-preset-chart-deckgl/src/layers

Ultimately, the H3 layer really just needs a key and a weight.  However then I looked at how miserable it appears to be to add a plug-in. Considering the experience we had getting Superset stably running and sending e-mail I was hoping it wouldn’t be as complex to add.

**Comment by jordibeen** [comment]
Would love this as well!

**Comment by terrorhawks** [comment]
@riazarbi Any news on supporting this feature, happy to help work on a PR

**Reply by riazarbi** [reply]
My day to day work no longer uses Superset so I don't really have the time to work on this. I still think it's a good idea though!

**Comment by Boreto1213** [comment]
I just opened a PR that introduces this: #38028
